### ISSUE TITLE
Hooks are not working

### ISSUE THREAD

**Issue by credli** [RobinBuschmann/sequelize-typescript#669]
**Versions**

- sequelize: 5.16.0
- sequelize-typescript: 1.0.0
- typescript: 3.5.3

**I'm submitting a ...** 

[X] bug report
[ ] feature request

**Actual behavior:**

Hooks `@BeforeCreate` and `@BeforeUpdate` are not triggered upon calling create on a model.

**Expected behavior:**

static function with hook annotation should work like specified in tests.

**Steps to reproduce:**

Create a model and add hooks to static function, run create on a model. The static function is not triggered.

**Related code:**

```ts
export enum LeadType {
  COMPANY,
  INDIVIDUAL,
}

@Table({
  freezeTableName: true,
  tableName: "Leads",
})
class Lead extends Model<Lead> {
  @BeforeCreate
  @BeforeUpdate
  public static makeLeadTitle(instance: Lead) {
    // this func is not called
    switch (instance.leadType) {
      case LeadType.COMPANY: {
        instance.leadTitle = instance.name;
      }
      case LeadType.INDIVIDUAL: {
        instance.leadTitle = `${instance.firstName} ${instance.lastName}`;
      }
    }
  }

  @PrimaryKey
  @IsUUID(4)
  @Column({
    type: DataType.UUIDV4,
    field: "ID",
  })
  public id?: string;

  @Column({
    type: DataType.ENUM,
    values: ["COMPANY", "INDIVIDUAL"],
    defaultValue: "COMPANY",
    field: "LeadType",
  })
  public leadType: LeadType;

  @Column({
    type: DataType.STRING,
    field: "Name",
  })
  public name?: string;

  @Column({
    type: DataType.STRING,
    field: "FirstName",
  })
  public firstName?: string;

  @Column({
    type: DataType.STRING,
    field: "LastName",
  })
  public lastName?: string;

  @Column({
    type: DataType.STRING,
    field: "LeadTitle",
  })
  public leadTitle: string;

  @Column({
    type: DataType.STRING,
    field: "Company",
  })
  public company?: string;
}
```

**Comment by credli** [comment]
Turned out this is not an issue with `sequelize-typescript` but instead it was related to an issue with enum switch. 

https://stackoverflow.com/questions/27747437/typescript-enum-switch-not-working
